jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/FilePropertiesStack.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli.buildfile;
+
+import com.google.cloud.tools.jib.api.buildplan.FilePermissions;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * A class that keeps track of permissions for various stacking file permissions settings in {@link
+ * LayerSpec}.
+ */
+@VisibleForTesting
+class FilePropertiesStack {
+
+  // TODO perhaps use a fixed size list here
+  private List<FilePropertiesSpec> stack = new ArrayList<>(3);
+
+  @Nullable private FilePermissions filePermissions;
+  @Nullable private FilePermissions directoryPermissions;
+  @Nullable private Instant modificationTime;
+  @Nullable private String ownership;
+
+  // intermediate values
+  @Nullable private String user;
+  @Nullable private String group;
```

Comment:
These don't seem necessary. In `updateProperties()`, you always start with `null`. These can be local to the method.

Reply:
Yeah, I can't remember why I did this, good catch.